src/us/kbase/workspace/database/provenance/PermanentID.java

```diff
@@ -0,0 +1,130 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import java.util.Optional;
+import us.kbase.workspace.database.Util;
+
+/**
+ * Represents a permanent unique identifier for an entity with an optional
+ * relationship to some other entity. That entity is expected to hold a
+ * reference to the PID.
+ */
+public class PermanentID {
+
+	private final String id;
+	private final String description;
+	private final String relationshipType;
+
+	private PermanentID(final String id, final String description, final String relationshipType) {
+		this.id = id;
+		this.description = description;
+		this.relationshipType = relationshipType;
+	}
+
+	/**
+	 * Gets the ID, for example DOI:10.25982/59912.37.
+	 *
+	 * @return the id.
+	 */
+	public String getID() {
+		return id;
+	}
+
+	/**
+	 * Gets a free text description of the resource identified by the ID.
+	 *
+	 * @return the description, if present.
+	 */
+	public Optional<String> getDescription() {
+		return Optional.ofNullable(description);
+	}
+
+	/**
+	 * Gets the relationship between the ID and some other entity. For
+	 * example, when a {@link PermanentID] class is used to represent
+	 * objects in a {@link Resource}'s 'related_identifiers' field, this
+	 * field captures the relationship between the {@link Resource} and
+	 * the entity represented by this.id.
+	 *
+	 * This field is currently only settable by workspace admins. More
+	 * validation may be added in the future to control the field contents.
+	 *
+	 * @return the relationship type, if present.
+	 */
+	public Optional<String> getRelationshipType() {
+		return Optional.ofNullable(relationshipType);
+	}
+
+	/**
+	 * Gets a builder for an {@link PermanentID}.
+	 *
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String id) {
```

Comment:
Needs `@param` entry for an id describing the format

Reply:
Done.